sdk/sync/barrier_test.go

```diff
@@ -0,0 +1,193 @@
+package sync
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+)
+
+func TestBarrier(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	closeFn := ensureRedis(t)
```

Comment:
I think it'd be better to call this in `TestMain`, so that we are sure we have Redis running for all tests. Otherwise if we run parallel tests, it is possible for a test to kill Redis, while it is used by another test.

Currently we run tests with `-p 1`, but it'd be nice to get rid of that.

Reply:
Yep, agreed! I only discovered TestMain a few weeks ago.